src/test/java/com/iexec/worker/sms/SmsServiceTests.java

```diff
@@ -51,15 +51,69 @@ void beforeEach() {
+        Signature signatureStub = new Signature(SIGNATURE);
         WorkerpoolAuthorization workerpoolAuthorization = mock(WorkerpoolAuthorization.class);
         when(credentialsService.hashAndSignMessage(workerpoolAuthorization.getHash()))
                 .thenReturn(signatureStub);
-        when(smsClient.createTeeSession(signatureStub.getValue(), workerpoolAuthorization))
-                .thenReturn(ResponseEntity.ok(SESSION_ID));
+        when(smsClient.generateTeeSession(signatureStub.getValue(), workerpoolAuthorization))
+                .thenReturn(SESSION_ID);
 
         String returnedSessionId = smsService.createTeeSession(workerpoolAuthorization);
-        assertThat(returnedSessionId).isEqualTo(SESSION_ID);
+        Assertions.assertThat(returnedSessionId).isEqualTo(SESSION_ID);
+    }
+
+    @Test
+    void shouldNotCreateTeeSessionOnFeignException() {
+        Signature signatureStub = new Signature(SIGNATURE);
+        WorkerpoolAuthorization workerpoolAuthorization = mock(WorkerpoolAuthorization.class);
+        when(credentialsService.hashAndSignMessage(workerpoolAuthorization.getHash()))
+                .thenReturn(signatureStub);
+        when(smsClient.generateTeeSession(signatureStub.getValue(), workerpoolAuthorization))
+                .thenThrow(FeignException.class)
+                .thenThrow(RuntimeException.class);
+        Assertions.assertThat(smsService.createTeeSession(workerpoolAuthorization)).isEmpty();
+        Assertions.assertThat(smsService.createTeeSession(workerpoolAuthorization)).isEmpty();
+        verify(smsClient, times(2))
+                .generateTeeSession(signatureStub.getValue(), workerpoolAuthorization);
+    }
+
+    @Test
+    void shouldGetSconeCasUrl() {
+        when(smsClient.getSconeCasUrl()).thenReturn(CAS_URL);
+        String sconeCasUrl = smsService.getSconeCasUrl();
+        Assertions.assertThat(sconeCasUrl).isEqualTo(CAS_URL);
+        verify(smsClient).getSconeCasUrl();
+    }
+
+    @Test
+    void shouldNotGetSconeCasUrlOnException() {
+        when(smsClient.getSconeCasUrl()).thenThrow(FeignException.class);
+        Assertions.assertThat(smsService.getSconeCasUrl()).isEmpty();
+        verify(smsClient).getSconeCasUrl();
+    }
+
+    @Test
+    void shouldGetTeeWorkflowConfiguration() {
+        TeeWorkflowSharedConfiguration teeWorkflowConfiguration = mock(TeeWorkflowSharedConfiguration.class);
+        when(smsClient.getTeeWorkflowConfiguration()).thenReturn(teeWorkflowConfiguration);
+        Assertions.assertThat(smsService.getTeeWorkflowConfiguration()).isEqualTo(teeWorkflowConfiguration);
+        verify(smsClient).getTeeWorkflowConfiguration();
+    }
+
+    @Test
+    void shouldGetTeeWorkflowConfigurationWhenAlreadyDefined() {
+        TeeWorkflowSharedConfiguration teeWorkflowConfiguration = mock(TeeWorkflowSharedConfiguration.class);
+        ReflectionTestUtils.setField(smsService, "teeWorkflowConfiguration", teeWorkflowConfiguration);
```

Comment:
To avoid reflection, we could call `smsService.getTeeWorkflowConfiguration()` twice and assert that `smsClient.getTeeWorkflowConfiguration()` is only called once.

Reply:
Fixed in bf57fed, thanks for the review